### PR TITLE
Clean up some phrasing, TODOs, etc.

### PR DESCRIPTION
**Pull request by ocsw**
**What this PR does / why we need it**:
Updates and fixes OWNERS
Adds janitor plugin summaries - was a TODO
Cleans up some capitalizations and wordings
Suppresses a test warning - seems to have moved, also made the suppression more specific

**Which issue(s) this PR fixes**

Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Updated and fixed OWNERS
Added janitor plugin summaries
Cleaned up some capitalizations and wordings
Suppressed a test warning
```

### CODE REVIEW

**Comment by ocsw** [comment]
@spotify/alf ok?

File: OWNERS [hunk 1]
```diff
@@ -2,5 +2,6 @@
   - descent3me
   - econchick
   - matt0bi
-  - RockesterinNYC
+  - RochesterinNYC
   - smstone
+  - ocsw
```
Comment (econchick) [inline OWNERS]: alpha order?

Comment (ocsw) [reply OWNERS]: +1

**Comment by ocsw** [comment]
@econchick ok?

File: tests/unit/plugins/service/test_init.py [hunk 2]
```diff
@@ -71,9 +71,10 @@ def exp_sub(consumer_config):
      'projects/test-example/subscriptions/a-subscription'),
     (False, False, 'projects/test-example/topics/a-topic', 'a-subscription'),
 ])
-def test_get_event_consumer(local, provide_loop, topic, sub, consumer_config,
-                            exp_topic, auth_client, exp_sub, subscriber_client,
-                            emulator, monkeypatch, event_loop):
+def test_get_event_consumer_WARNING_EXPECTED(
```
Comment (econchick) [inline tests/unit/plugins/service/test_init.py]: I would rather filter out an expected warning than have it in the name (see [pytest docs](https://docs.pytest.org/en/latest/warnings.html#pytest-mark-filterwarnings)).

File: README.rst [hunk 3]
```diff
@@ -10,11 +10,17 @@ Google Cloud Platform (GCP) plugins for `gordon`_, an open-source, event-driven
...
-* Creating, updating, and deleting records within `Google Cloud DNS`_
+Service:
+
 * Consuming events from `Google Cloud Pub/Sub`_
 * Reading from `Google Compute Engine`_ for record information
+* Creating, updating, and deleting records within `Google Cloud DNS`_
+
+Janitor:
 
-TODO (zephyr): what do the janitor plugins do?
```
Comment (econchick) [inline README.rst]: Sorry I missed this before. But in the future for "todos" in prose documentation, we use the [todo extention](http://www.sphinx-doc.org/en/master/ext/todo.html) ([example of our usage](https://raw.githubusercontent.com/spotify/gordon/develop/docs/plugins.rst)).

File: OWNERS [hunk 4]
```diff
@@ -2,5 +2,6 @@
   - descent3me
   - econchick
   - matt0bi
-  - RockesterinNYC
```
Comment (econchick) [inline OWNERS]: 🤘

File: src/gordon_gcp/plugins/janitor/__init__.py [hunk 5]
```diff
@@ -32,18 +32,18 @@
 
 
 def get_gpubsub_publisher(config, changes_channel, **kw):
-    """Get a GPubsubPublisher client.
+    """Get a GPubsubPublisher instance.
 
     A factory function that validates configuration, creates an auth
     and pubsub API client, and returns a Google Pub/Sub Publisher
-    provider.
```
Comment (econchick) [inline src/gordon_gcp/plugins/janitor/__init__.py]: Hmm "provider" is specific to interface implementation, but I understand the switch to 'instance'. What do you think of: 

> ...and returns a Google Pub/Sub Publisher instance, a provider of the ``IPublisherClient`` interface.

File: src/gordon_gcp/plugins/janitor/gpubsub_publisher.py [hunk 6]
```diff
@@ -14,19 +14,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """
-Client module to publish any required DNS changes initiated from
+Plugin module to publish any required DNS changes initiated from
```
Comment (econchick) [inline src/gordon_gcp/plugins/janitor/gpubsub_publisher.py]: Hm "client" v "plugin" terminology - I think I'd like someone else ( @matt0bi @descent3me ?) to chime in w/ their opinion: the interfaces are defined as `IPublisherClient`, `IEnricherClient`, etc. They are gordon "clients" to interact with Google products.

I can see the confusion around a gordon "client" (aka implemented interfaces / interface providers) and the API clients we've also defined (`GCEClient`, `GDNSClient`, etc).

Basically, I'd vouch for consistency. So either we keep it as "client" here, or we change the names of the interface definitions.

Comment (ocsw) [reply src/gordon_gcp/plugins/janitor/gpubsub_publisher.py]: I was actually going to suggest we change those.

File: src/gordon_gcp/plugins/janitor/__init__.py (hunk 5 above)
Comment (ocsw) [reply src/gordon_gcp/plugins/janitor/__init__.py]: WFM

File: src/gordon_gcp/plugins/janitor/gpubsub_publisher.py (hunk 6 above)
Comment (descent3me) [reply src/gordon_gcp/plugins/janitor/gpubsub_publisher.py]: I'm okay with renaming the interfaces to be plugins, and think that for consistency we shouldn't change the docs until the code changes.

Comment (matt0bi) [reply src/gordon_gcp/plugins/janitor/gpubsub_publisher.py]: I'm for switching the phrasing from client to plugin here - I can see the argument that these are plugin classes (+ they're in the "plugins" directory now).

**Comment by ocsw** [comment]
@econchick so, ok if I remove that one commit, merge this, and then start changing gordon/gordon-janitor and work my way down?

**Comment by econchick** [comment]
@ocsw removing that one commit to merge, yeah. But IMO I think the work for changing the wording on everything should be made into a ticket and worked on later since it's not a high priority (maybe @descent3me + @matt0bi think otherwise though).

**Comment by ocsw** [comment]
Ok, pulled out that stuff into a new branch. @econchick ok?

File: tests/unit/plugins/service/test_init.py (hunk 2 above)
Comment (ocsw) [reply tests/unit/plugins/service/test_init.py]: Seems like the spot generating the warning moved?  Fixed now.

**Comment by ocsw** [comment]
@econchick also fixed warning issue

File: tests/unit/plugins/service/test_init.py [hunk 7]
```diff
@@ -61,19 +61,15 @@ def exp_sub(consumer_config):
...
 @pytest.mark.parametrize('local,provide_loop,topic,sub', [
     (True, True, 'a-topic',
      'projects/test-example/subscriptions/a-subscription'),
     (False, False, 'projects/test-example/topics/a-topic', 'a-subscription'),
 ])
-def test_get_event_consumer(local, provide_loop, topic, sub, consumer_config,
-                            exp_topic, auth_client, exp_sub, subscriber_client,
-                            emulator, monkeypatch, event_loop):
+def test_get_event_consumer(
+        local, provide_loop, topic, sub, consumer_config, exp_topic,
```
Comment (econchick) [inline tests/unit/plugins/service/test_init.py]: This change from a linter? This style change goes against the rest of the style of new lines for func args.

File: tests/unit/plugins/service/test_gdns_publisher.py [hunk 8]
```diff
@@ -286,6 +286,13 @@ def mock_sleep(mocker):
 #####
 # GDNS Publisher Tests
 #####
+# For some reason, the event loop for this test leaks over to
+# tests.unit.plugins.service.test_event_consumer:test_gpsthread_add_task so
+# there's a warning of not awaiting the coroutine (which is expected).
+# Threads + asyncio + testing is hard.
```
Comment (econchick) [inline tests/unit/plugins/service/test_gdns_publisher.py]: It's still hard. gah.

Comment (matt0bi) [reply tests/unit/plugins/service/test_gdns_publisher.py]: Lol, so true @econchick.
